component/pyroscope/java/args.go

```diff
@@ -0,0 +1,43 @@
+package java
+
+import (
+	"time"
+
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/component/pyroscope"
+)
+
+type Arguments struct {
+	Targets   []discovery.Target     `river:"targets,attr"`
+	ForwardTo []pyroscope.Appendable `river:"forward_to,attr"`
+
+	TmpDir          string          `river:"tmp_dir,attr,optional"`
```

Comment:
Yeah that's a good idea !

Reply:
@tpaschalis Does the `options.DataPath` is unique per component or do we need to create a folder for our component ?

Reply:
The path should be the same on host and container, so the `pyroscope.java` component may recreate directories and I dont really want it to touch anything except /tmp. I already dont like the fact that if `/tmp` does not exist in the target container, the `pyroscope.java` component is currently creating it with 0755 permissions and without sticky bit.
It also should have read permissions for unprivileged users on host, so that unprivileged java process on host (not container, same mount ns) can read and load `libAsyncProfiler.so`.

The path should also be not writable by unprivileged user. We unpack an elf binary to `/tmp/grafana-agent-.../bin/asprof` and then invoke it as privileged user.  So we better make sure unprivileged user can not modify or manipulated it in any way(symlinks or incorrect permissions)

I've tried to implement these checks with `/tmp` having sticky bit, but still not 100% sure it is fully correct.